### DISCUSSION TITLE
No serial Port found

### DISCUSSION THREAD

**Discussion by JoseMoranRodriguez** [winder/Universal-G-Code-Sender#1820] [Help]
Hello everyone my name is Jose, and I’m newbie on run my mill machine with a OpenSource software like UGS, so I tried run my mill but when I plug my machine and choose the firmware: GRBL then look for any Port the error comes with the next message: “No serial ports found”. The model of my CNC board is BL-USBMach-2.1 and when I ran my mill on Mach3 it didn’t exist any problem, I could control the machine.

I thought the problem is the driver but I didn’t find any driver for that, besides on Mach3 didn’t need driver only the plugin with USB. I tried in two Operative Systems like Linux and Windows, the problem still remain.

So any ideas to resolve this problem, I would be so grateful!

**Comment by chrisla23** [comment]
Are you sure your board is supported? UGS primarily works with the open-source grbl controller.  A few others are also included, but I don't think this is one of them.

Thanks,

-Chris

**Reply by JoseMoranRodriguez** [reply]
I'm not sure If is my board is supported, any recommendation? Does not exist any way to connect my board on open source grbl controller?

**Comment by chrisla23** [comment]
Likely anything is possible, but if you wanted to convert your machine to grbl, you would need an Arduino and a Stepper driver interface board. You would essentially replace your board with a grbl board. The stepper motors would connect to that. 

I am curious what your motivations are? Mach 3 is great, and was/is the defacto standard in CNC. I can't say I may not use it myself if my machine supported it. 

If it is to save on a license you will probably burn more time and cost trying to make this work than a license costs. People who use this are doing so since their machine came that way or are building from scratch. 

Thanks,

-Chris
